### DISCUSSION TITLE
Need help with Installing Monailabel

### DISCUSSION THREAD

**Discussion by jankh0** [Project-MONAI/MONAILabel#1788] [General]
Hi Everyone,

I a researcher trying to use monailabel to segment the features of the neck including vasculature, tissue, trachea and muscle. I'm having significant problems with installation and have tried number of permutations including the below.

I am installing it on a windows 11 machine using terminal on VSCode. 

1. https://projectweek.na-mic.org/PW37_2022_Virtual/Projects/MONAILabelLung/MONAILabel_Installation.html
2. https://www.youtube.com/watch?v=8y1OBQs2wis
3. https://discourse.slicer.org/t/how-to-start-with-monailabel-for-new-models/21063/65?page=4 - instructions.

I seem to keep running into dll errors, path errors and CUDA cores not detected on Torch despite updated drivers 

I've tried Conda and when I am in a Conda environment, the files appear to be missing also. 

Are there installation instructions elsewhere that may potentially help? 

Kind Regards

Jan

**Comment by SachidanandAlle** [comment]
This is more of an environment/system issue for cuda rather than monailabel.

If you have cuda is correctly installed and you should be able to run `nvidia-smi` command successfully.

You can refer PyTorch installation guide to install torch with cuda support 
https://pytorch.org/get-started/locally/

Cond/PIP or anything, if you can make `python -c "import torch; print(torch.cuda.is_available())"` working, you can install monailabel.

**Comment by jankh0** [comment]
Hi SachidanandAlle, 

Thank you for taking the time to reply me. 

I've done the below. 

       pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu113
      
       python -c "import torch; print(torch.cuda.is_available())" 
      FALSE
      
       nvidia-smi 

        +-----------------------------------------------------------------------------------------+
        | NVIDIA-SMI 566.14                 Driver Version: 566.14         CUDA Version: 12.7     |
        |-----------------------------------------+------------------------+----------------------+
        | GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
        | Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
        |                                         |                        |               MIG M. |
        |=========================================+========================+======================|
        |   0  NVIDIA GeForce RTX 4090 ...  WDDM  |   00000000:01:00.0  On |                  N/A |
        | N/A   45C    P8              8W /   81W |    2103MiB /  16376MiB |      4%      Default |
        |                                         |                        |                  N/A |
        +-----------------------------------------+------------------------+----------------------+

Perhaps I may not have installed the appropriate pytorch. 

      pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118

      Requirements satisfied. 


I'm not sure what else needs to be done for cuda availabe to return True. 

Thanks again for your help.

**Reply by diazandr3s** [reply]
Hi @jankh0,

Thanks for sharing these details.
You may find this helpful: https://stackoverflow.com/questions/60987997/why-torch-cuda-is-available-returns-false-even-after-installing-pytorch-with

Taken from this post:

> The system requirements to use PyTorch with CUDA are as follows:
> 
> 1/ Your graphics card must support the required version of CUDA
> 2/ Your graphics card driver must support the required version of CUDA
> 3/ The PyTorch binaries must be built with support for the compute capability of your graphics card

**Comment by jankh0** [comment]
Hi @diazandr3s and @SachidanandAlle,

Thank you for the replies. I've managed to get monailabel started after some troubleshooting.

I'm not sure if I'm speaking out of turn but installation was not as straightforward and ChatGPT managed to get me past the problems. 

Kind Regards

Jan
